packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -433,6 +623,196 @@ class _CircularProgressIndicatorPainter extends CustomPainter {
+///           value: 0.5,
+///           semanticsLabel: 'Circular progress indicator',
+///         ),
+///       ],
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_material_ticker}
+///
+/// This example shows a [CircularProgressIndicator] with a changing color.
+///
+/// ```dart
+/// AnimationController controller;
+/// Animation<Color> animation;
+///
+/// @override
+/// void initState() {
+///   controller = AnimationController(
+///     vsync: this,
+///     duration: const Duration(seconds: 5),
+///   );
+///   animation = ColorTween(
+///     begin: Colors.green,
+///     end: Colors.blue,
+///   ).animate(controller);
+///   controller.repeat(reverse: true);
+///   super.initState();
+/// }
+///
+/// @override
+/// void dispose() {
+///   controller.dispose();
+///   super.dispose();
+/// }
+///
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     body: Padding(
+///       padding: const EdgeInsets.all(20.0),
+///       child: Column(
+///         mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+///         children: [
+///           Text(
+///             'Circular progress indicator with a fixed color',
+///             style: const TextStyle(fontSize: 20),
+///           ),
+///           CircularProgressIndicator(
+///             valueColor: animation,
+///             semanticsLabel: 'Circular progress indicator',
+///           ),
+///         ],
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_material_ticker}
+///
+/// This example shows a [CircularProgressIndicator] with a changing value.
+///
+/// ```dart
+/// AnimationController controller;
+/// Animation<double> animation;
+///
+/// @override
+/// void initState() {
+///   controller = AnimationController(
+///     vsync: this,
+///     duration: const Duration(seconds: 5),
+///   );
+///   animation = Tween<double>(
+///     begin: 0,
+///     end: 1,
+///   ).animate(controller)
+///     ..addListener(() {
+///       setState(() {});
+///     });
+///   controller.repeat(reverse: true);
+///   super.initState();
+/// }
+///
+/// @override
+/// void dispose() {
+///   controller.dispose();
+///   super.dispose();
+/// }
+///
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     body: Padding(
+///       padding: const EdgeInsets.all(20.0),
+///       child: Column(
+///         mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+///         children: [
+///           Text(
+///             'Linear progress indicator with a fixed color',
+///             style: const TextStyle(fontSize: 20),
+///           ),
+///           CircularProgressIndicator(
+///             value: animation.value,
+///             semanticsLabel: 'Linear progress indicator',
+///           ),
+///         ],
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
```

Comment:
Same here, each of these samples are great, but we probably only need one DartPad sample for this widget

Reply:
Should I do the same as above? (Combine the fixed color and value)

Reply:
Yes, could you trim this down to one sample for LinearProgressindicator and one for CircularProgressIndicator?

Reply:
I have already added a commit fixing this. 
Thanks

Reply:
It looks like there's still more than we need:

1. Linear progressIndicator with a fixed value and color
2. LinearProgressIndicator with a fixed color, fixed value, and a background color
3. LinearProgressIndicator DartPad sample with a changing color
4. LinearProgressIndicator DartPad with a changing value
5. CircularProgressIndicator with a fixed value and color
6. CircularProgressIndicator with a fixed color, fixed value, and a background color
7. CircularProgressIndicator DartPad with a changing color
8. CircularProgressIndicator DartPad with a changing value

I think we should to keep 4. and 8 and remove the rest. Too many samples can make it harder for a reader to find what they're looking for.